### PR TITLE
shortcut for invertability check

### PR DESCRIPTION
**Pull request by abh2k**
* matrices
  *  non-square matrices now raise NonSquareMatrixError if inversion is attempted

### CODE REVIEW

**Comment by abhinav28071999** [comment]
I will implement shortcut for other inverse methods also in this pr.

File: sympy/matrices/inverse.py [hunk 1]
```diff
@@ -261,7 +261,10 @@ def _inv_LU(M, iszerofunc=_iszero):
     inverse_LDL
     """
 
-    _verify_invertible(M, iszerofunc=iszerofunc)
+    if not M.is_square:
+        raise NonSquareMatrixError("A Matrix must be square to invert.")
+    if M.free_symbols != set():
```
Comment (smichr) [inline sympy/matrices/inverse.py]:
```suggestion
    if M.free_symbols:
```

**Comment by smichr** [comment]
Let's see if anyone has feedback about raising this error instead of the previous.

**Comment by sylee957** [comment]
I’m afraid that many of the sympy decomposition methods are generalized to singular matrices, and I wonder if the solvers raise errors properly for non-invertible matrix or they are giving NaN values.
This should be confirmed

**Comment by abhinav28071999** [comment]
Okay, i will test this for some non invertible matrices.

**Comment by czgdp1807** [comment]
LGTM.

**Comment by czgdp1807** [comment]
Though, please merge `master`. `Codecov` seems to use something else as diff.

**Comment by abhinav28071999** [comment]
@sylee957 @jksuom @oscarbenjamin @smichr https://en.wikipedia.org/wiki/Rank_(linear_algebra)#Computation
We can see it is possible to compute rank of matrix during LU decomposition and if rank_row != rank_col that means the det should be 0. Which  this pr is doing. I have tested  with many examples and is working.

**Pull request merged**